lib/trino-parquet/src/main/java/io/trino/parquet/writer/PrimitiveColumnWriter.java

```diff
@@ -309,7 +309,7 @@ private DataStreams getDataStreams()
         getDataStreamsCalled = true;
```

Comment:
> Bloom filter writing logic is changed to always collect a bloom filter and
discard it at the end if dictonary is present

So dictionaries and bloom filters are exclusive? Is there a potential perf degradation?

> Current logic was failing to convert dictionary pages into bloom filter
when the fallback from dictionary to plain encoding happened after first page.

Do we know the reason.

Reply:
The current logic assumed fallback would only happen on the first page/there would only be one page.

Reply:
> So dictionaries and bloom filters are exclusive?

Initial PR was a bit too restrictive, i've tweaked it a bit to discard bloom filter only if all pages are dictionary encoded, rather than just on presence of dictionary.

> Is there a potential perf degradation?

We will pay extra cost of building bloom filter even if the column turns out to be fully dictionary encoded. This will be only on bloom filter configured columns though.

The problem with the fallback based approach was that it would be more complicated to make ValueWriter remember all previous dictionary encoded pages and decode them back to plain values for populating into bloom filter. It's simpler to just populate bloom filter separately and discard if it's redundant with dictionary.